actions/copy/copy.js

```diff
@@ -17,34 +17,58 @@
 
 // eslint-disable-next-line import/no-extraneous-dependencies
 const openwhisk = require('openwhisk');
-const { getAioLogger } = require('../utils');
+const {
+    getAioLogger, updateStatusToStateLib, COPY_ACTION
+} = require('../utils');
 
 // This returns the activation ID of the action that it called
 function main(args) {
     const logger = getAioLogger();
     let payload;
+    const {
+        spToken, adminPageUri, projectExcelPath, projectRoot
+    } = args;
+    const projectPath = `${projectRoot}${projectExcelPath}`;
     try {
-        const { spToken, adminPageUri, projectExcelPath } = args;
-        if (!spToken || !adminPageUri || !projectExcelPath) {
+        if (!projectRoot || !projectExcelPath) {
+            payload = 'Could not determine the project path. Try reloading the page and trigger the action again.';
+            logger.error(payload);
+        } else if (!spToken || !adminPageUri) {
             payload = 'Required data is not available to proceed with FG Promote action.';
+            updateStatusToStateLib(projectPath, 'Failure', payload, '', COPY_ACTION);
             logger.error(payload);
         } else {
+            updateStatusToStateLib(projectPath, 'In-Progress', 'Triggering copy action', '', COPY_ACTION);
```

Comment:
check if a project is already running before triggering the call.

Reply:
@sukamat  Please check this now. Added a check before triggering a new action on the worker.